js/state/syncUtil.js

```diff
@@ -242,7 +243,7 @@ module.exports.getExistingObject = (categoryName, syncRecord) => {
       item = module.exports.createSiteData(existingObjectData, appState)
       break
     case 'PREFERENCES':
-      const hostPattern = existingObject[0]
+      const hostPattern = existingObjectKeyPath[existingObjectKeyPath.length - 1]
```

Comment:
an existing object's app state key path should be an array of at least length 1. so for siteSettings: ["siteSettings", {hostPattern}]